be1-go/message/generic_message.go

```diff
@@ -47,3 +47,17 @@ func (m *GenericMessage) UnmarshalJSON(data []byte) error {
 	m.Query = query
 	return nil
 }
+
+type IdStruct struct {
```

Comment:
Sorry I overlooked this before, you might want to move this inside `UnmarshalID` and declare it as a private struct

Reply:
No worries, I made the changes :)